src/gui/transferlistsortmodel.cpp

```diff
@@ -39,6 +39,44 @@
 
 namespace
 {
+    bool operator<(const QVariant &left, const QVariant &right)
+    {
+        Q_ASSERT(left.userType() == right.userType());
+        if (left.userType() != right.userType()) [[unlikely]]
+            return false;
```

Comment:
Seems this is redundant. QVariant `to*` methods will still return some (invalid) value that is usable in comparison.
```suggestion
```

Reply:
> Seems this is redundant. QVariant `to*` methods will still return some (invalid) value that is usable in comparison.

This method isn't intended to be used with values of different types so I would keep assertion in order to detect possible misuses.
Note that I am not offering it for general use. Its scope is limited only to this file.

Reply:
>This method isn't intended to be used with values of different types so I would keep assertion in order to detect possible misuses.

To be clear, I'm talking about the `if()` statement, not the `assert`.